packages/accounts-base/accounts_server.js

```diff
@@ -77,9 +77,10 @@ export class AccountsServer extends AccountsCommon {
     // user expects. The way to make this work in a publish is to do
     // Meteor.find(this.userId).observe and recompute when the user
     // record changes.
-    var currentInvocation = DDP._CurrentInvocation.get();
+    var currentInvocation = DDP._CurrentPublicationInvocation.get();
+    currentInvocation = currentInvocation || DDP._CurrentInvocation.get();
     if (!currentInvocation)
-      throw new Error("Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.");
+      throw new Error("Meteor.userId can only be invoked in method calls or publications.");
```

Comment:
Can you also change "This function only works if called inside a method." comment above. And maybe change the rest of the comment to warn that if you use `Meteor.user()` (not `Meteor.userId()`) outside of the reactive context on the server the data returned by `Meteor.user()` will not change reactivelly. I mean, to me this is pretty obvious, so maybe we should just remove this comment.